hibernate-core/src/main/java/org/hibernate/cfg/AnnotationBinder.java

```diff
@@ -3113,7 +3113,12 @@ else if (hasSpecjManyToOne) {
 		propertyBinder.setCascade( cascadeStrategy );
 		propertyBinder.setProperty( property );
 		propertyBinder.setXToMany( true );
-		propertyBinder.makePropertyAndBind();
+		if ( property.isAnnotationPresent( JoinTable.class ) ) {
+			propertyBinder.makePropertyAndBind().setOptional( optional );
+		}
+		else {
+			propertyBinder.makePropertyAndBind();
```

Comment:
Any particular reason why this clause differs from the case with the `JoinTable`? It would seem only sensible that the optionality is always relevant for the bound property.  I trust that it is not, but then I am somewhat surprised that this is an appropriate fix 😉

Reply:
This patch is a bit different than the one I wrote, i'll look at it.

Reply:
A quick look at it, the bindManyToOne method may be used outside of the JoinTable situation, so setting optional would be disallowed. Also not all usage of JoinTable gets you to this code, there has to be a OneToOne.

Reply:
Hi @jwgmeligmeyling I'm looking at the issue again, and I tend to agree with you. 

The problem with just setting `propertyBinder.makePropertyAndBind().setOptional( optional );` is for cases like the one in `NonNullableCircularDependencyCascadeTest` where we have 
```
@Entity
public class Child
{
   ...

   @ManyToOne(cascade=CascadeType.PERSIST)
   @JoinColumn(name="parentId", nullable=false)
	Parent parent; 
    ...
}
```

I'm pushed a slightly different solution.

Reply:
Looks good to me!

Reply:
Thanks @jwgmeligmeyling for the review